src/components/ArticlePage.jsx

```diff
@@ -8,14 +8,22 @@ const ArticlePage = () => {
     const {article_id} = useParams()
     const [article, setArticle] = useState()
     const [isLoading, setIsLoading] = useState(true)
+    const [votes, setVotes] = useState()
 
     useEffect(() => {
         getArticleById(article_id).then((res) => {
             setArticle(res.data.article)
+            setVotes(res.data.article.votes)
             setIsLoading(false)
         })
     }, [])
 
+    function handleVote(article_id, incVote) {
+        patchArticle(article_id, incVote)
+        const newVotes = votes + incVote
+        setVotes(newVotes)
+    }
```

Comment:
A nice start to optimistic rendering implementation, but what if the upvote doesn't go through or there's an error - we should then return the state to what it was before, have a look at these notes for more info https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#error-handling